runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/DefaultIllumina.java

```diff
@@ -157,12 +183,15 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
       throw new IOException(e);
     }
 
-    // See if we can figure out the chemistry
+    Document runParams = Stream.of("runParameters.xml", "RunParameters.xml").map(f -> new File(runDirectory, f))
```

Comment:
You can keep this as `Optional<Document>` and then replace the `if` block with an `ifPresent` method.